q2_fondue/metadata.py

```diff
@@ -35,71 +34,65 @@ def _efetcher_inquire(
     Args:
         efetcher (ef.Efetcher): A valid instance of an Entrezpy Efetcher.
         run_ids (List[str]): List of all the run IDs to be fetched.
+        log_level (str): Logging level.
 
     Returns:
         pd.DataFrame: DataFrame with metadata obtained for the provided IDs.
-        list: List of all the run IDs that were not found.
+        dict: Dictionary of the run IDs that were not found with
+            respective error messages.
     """
     metadata_response = efetcher.inquire(
         {
             'db': 'sra',
             'id': run_ids,
             'rettype': 'xml',
             'retmode': 'xml',
-            'retmax': BATCH_SIZE
+            'retmax': len(run_ids),
+            'reqsize': BATCH_SIZE
         }, analyzer=EFetchAnalyzer(log_level)
     )
 
     if metadata_response.result is None:
         return (pd.DataFrame(),
                 {m_id: metadata_response.error_msg for m_id in run_ids})
     else:
-        return (metadata_response.result.metadata_to_df(), {})
-
-
-def _execute_efetcher(email, n_jobs, run_ids, log_level, logger):
-    meta_df = []
-    missing_ids = {}
-    for num, batch in enumerate(_chunker(run_ids, BATCH_SIZE), 1):
-        # one efetcher object per loop because threads of one
-        # efetcher object can only be started once:
-        efetcher = ef.Efetcher(
-            'efetcher', email, apikey=None,
-            apikey_var=None, threads=n_jobs, qid=None
-        )
-        set_up_entrezpy_logging(efetcher, log_level)
+        return metadata_response.result.metadata_to_df(), {}
 
-        logger.info(
-            f'Fetching metadata of run IDs from batch number {num}...'
-        )
-        df, missing_ids = _efetcher_inquire(efetcher, batch, log_level)
 
-        meta_df.append(df)
-        missing_ids.update(missing_ids)
+def _execute_efetcher(email, n_jobs, run_ids, log_level):
+    efetcher = ef.Efetcher(
+        'efetcher', email, apikey=None,
+        apikey_var=None, threads=n_jobs, qid=None
+    )
+    set_up_entrezpy_logging(efetcher, log_level)
 
-    return pd.concat(meta_df, axis=0), missing_ids
+    return _efetcher_inquire(efetcher, run_ids, log_level)
 
 
 def _get_run_meta(
-        email, n_jobs, run_ids, log_level, logger
+        email, n_jobs, run_ids, validated, log_level, logger
 ) -> (pd.DataFrame, dict):
-    invalid_ids = _validate_run_ids(email, n_jobs, run_ids, log_level)
-    valid_ids = sorted(list(set(run_ids) - set(invalid_ids.keys())))
-
-    if not valid_ids:
-        raise InvalidIDs(
-            'All provided IDs were invalid. Please check your input.'
-        )
-    if invalid_ids:
-        logger.warning(
-            f'The following provided IDs are invalid: '
-            f'{",".join(invalid_ids.keys())}. Please correct them and '
-            f'try fetching those independently.'
-        )
+    if not validated:
```

Comment:
If we got those `run_ids` IDs from a BioProject or any other "group" ID there is no need to validate them again, hence this new param here.

Reply:
where exactly were the run IDs validated in case of fetching them with the pipeline?

Reply:
As discussed irl, I'm leaving it as-is as there is no reason to believe that the IDs returned by NCBI (in the process of linking from aggregate to unique run IDs) would be invalid. I made a comment below and tried testing some invalid run IDs - I cannot recover their original BioProject IDs simply because they return no results when I search for those in any of the NCBI databases (which again indicates that whatever NCBI returns must be valid by default).